doc/api/fs.md

```diff
@@ -3215,6 +3215,763 @@ changes:
+and the file position will be updated.
+If `position` is an integer, the file position will remain unchanged.
+
+Following successful read, the `Promise` is resolved with an object with a
+`bytesRead` property specifying the number of bytes read, and a `buffer` property
+that is a referece to the passed in `buffer` argument.
+
+### fs.promises.readdir(path[, options])
+<!-- YAML
+added: REPLACEME
+-->
+
+* `path` {string|Buffer|URL}
+* `options` {string|Object}
+  * `encoding` {string} **Default:** `'utf8'`
+* Returns: {Promise}
+
+Reads the contents of a directory then resolves with `Promise` with an array
+of the names of the files in the directory excludiing `'.'` and `'..'`.
+
+The optional `options` argument can be a string specifying an encoding, or an
+object with an `encoding` property specifying the character encoding to use for
+the filenames passed to the callback. If the `encoding` is set to `'buffer'`,
+the filenames returned will be passed as `Buffer` objects.
+
+### fs.promises.readFile(path[, options])
+<!-- YAML
+added: REPLACEME
+-->
+
+* `path` {string|Buffer|URL|[FileHandle][]} filename or `FileHandle`
+* `options` {Object|string}
+  * `encoding` {string|null} **Default:** `null`
+  * `flag` {string} **Default:** `'r'`
+
+Asynchronously reads the entire contents of a file.
+
+The `Promise` is resolved with the contents of the file. If no encoding is
+specified (using `options.encoding`), the data is returned as a `Buffer`
+object. Otherwise, the data will be a string.
+
+If `options` is a string, then it specifies the encoding.
+
+When the `path` is a directory, the behavior of `fs.promises.readFile()` is
+platform-specific. On macOS, Linux, and Windows, the promise will be rejected
+with an error. On FreeBSD, a representation of the directory's contents will be
+returned.
+
+Any specified `FileHandle` has to support reading. If a `FileHandle` is
+specified as the `path`, it will not be closed automatically.
+
+### fs.promises.readlink(path[, options])
+<!-- YAML
+added: REPLACEME
+-->
+
+* `path` {string|Buffer|URL}
+* `options` {string|Object}
+  * `encoding` {string} **Default:** `'utf8'`
+* Returns: {Promise}
+
+Asynchronous readlink(2). The `Promise` is resolved with the `linkString` upon
+success.
+
+The optional `options` argument can be a string specifying an encoding, or an
+object with an `encoding` property specifying the character encoding to use for
+the link path passed to the callback. If the `encoding` is set to `'buffer'`,
+the link path returned will be passed as a `Buffer` object.
+
+### fs.promises.realpath(path[, options])
+<!-- YAML
+added: REPLACEME
+-->
+
+* `path` {string|Buffer|URL}
+* `options` {string|Object}
+  * `encoding` {string} **Default:** `'utf8'`
+* Returns: {Promise}
+
+Determines the actual location of `path` using the same semantics as the
+`fs.realpath.native()` function then resolves the `Promise` with the resolved
```

Comment:
No, I don't think so. The current `fs.realpath()` function is a legacy artifact that we're pretty much stuck with thanks to existing userland code. Since this is a new API, we have the opportunity not to continue that forward.